### PR TITLE
CDD-1831: Part1 - Extracted authentication logic to new CustomsAuthSe…

### PR DESCRIPTION
**Pull request by googley42**
…rvice so that client code can focus on business logic.

Remaining work planned for Part2:
- migrate AuthStatusAction to use new service
- move auth Header Validation functions to HeaderValidation service

### CODE REVIEW

File: app/uk/gov/hmrc/customs/declaration/controllers/actionbuilders/AuthAction.scala [hunk 1]
```diff
@@ -57,199 +50,66 @@ import scala.util.control.NonFatal
...
-        }
-    }.recover{
-      case NonFatal(_: AuthorisationException) =>
-        googleAnalyticsConnector.failure(errorResponseUnauthorisedGeneral.message)
-        Left(errorResponseUnauthorisedGeneral)
-      case NonFatal(e) =>
-        logger.error("Error authorising non-CSP", e)
-        Left(ErrorInternalServerError)
+    maybeBadgeId.filter(xBadgeIdentifierRegex.findFirstIn(_).nonEmpty).map(BadgeIdentifier).toRight[ErrorResponse]{
+      logger.error("badge identifier invalid or not present for CSP")
```
Comment (rashadmughal) [inline app/uk/gov/hmrc/customs/declaration/controllers/actionbuilders/AuthAction.scala]: Can you use the header name instead? e.g.
`logger.error(s"$XBadgeIdentifierHeaderName invalid or not present for CSP")`

File: app/uk/gov/hmrc/customs/declaration/services/CustomsAuthService.scala [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+      case NonFatal(e) =>
+        logger.error("Error authorising non-CSP", e)
+        Left(ErrorInternalServerError)
+    }
+  }
+
+  private def findEoriInCustomsEnrolment[A](enrolments: Enrolments, authHeader: Option[Authorization])(implicit vhr: ValidatedHeadersRequest[A], hc: HeaderCarrier): Option[Eori] = {
+    val maybeCustomsEnrolment = enrolments.getEnrolment("HMRC-CUS-ORG")
+    if (maybeCustomsEnrolment.isEmpty) {
+      logger.warn(s"Customs enrolment HMRC-CUS-ORG not retrieved for authorised non-CSP call")
```
Comment (rashadmughal) [inline app/uk/gov/hmrc/customs/declaration/services/CustomsAuthService.scala]: HMRC-CUS-ORG can be extracted into a val now that its being used in a few places

File: app/uk/gov/hmrc/customs/declaration/controllers/actionbuilders/AuthAction.scala [hunk 3]
```diff
@@ -17,36 +17,29 @@
...
 import scala.util.Left
-import scala.util.control.NonFatal
+
 
 /** Action builder that attempts to authorise request as a CSP or else NON CSP
   * <ul>
   * <li/>INPUT - `ValidatedHeadersRequest`
   * <li/>OUTPUT - `AuthorisedRequest` - authorised will be `AuthorisedAs.Csp` or `AuthorisedAs.NonCsp`
-  * <li/>ERROR -
+  * <li/>ERROR -]
```
[the remark this answers is not in the document]
Comment (googley42) [reply app/uk/gov/hmrc/customs/declaration/controllers/actionbuilders/AuthAction.scala]: done

File: app/uk/gov/hmrc/customs/declaration/services/CustomsAuthService.scala (hunk 2 above)
Comment (googley42) [reply app/uk/gov/hmrc/customs/declaration/services/CustomsAuthService.scala]: done

**Pull request merged**